### ISSUE TITLE
Cant build with video

### ISSUE THREAD

**Issue by crashcoq** [pjsip/pjproject#4041]
### Describe the bug

I can't make project PSIP with video on debian (works without video).
I don't have windows and directshow

### Steps to reproduce

sudo ./configure CFLAGS="-fPIC"
echo "#define PJMEDIA_HAS_VIDEO		1" >> pjlib/include/pj/config_site.h
echo "#define PJMEDIA_HAS_OPENH264_CODEC	1" >> pjlib/include/pj/config_site.h
echo "#define PJMEDIA_VIDEO_DEV_HAS_SDL	1" >> pjlib/include/pj/config_site.h
sudo make dep
sudo make

### PJSIP version

2.11

### Context

Package installed:
swig libopencore-amrwb-dev libopencore-amrnb-dev libvpx-dev libx264-dev libsdl2-dev libv4l-dev libopenh264-dev

file pjlib/include/pj/compat/os_auto.h
* Canonical OS name */
#define PJ_OS_NAME "armv7l-unknown-linux-gnueabihf"

/* Legacy macros */
/* #undef PJ_WIN64 */
/* #undef PJ_WIN32 */
/* #undef PJ_WIN32_WINNT */
/* #undef WIN32_LEAN_AND_MEAN */
/* #undef PJ_DARWINOS */
#define PJ_LINUX 1
/* #undef PJ_RTEMS */
/* #undef PJ_SUNOS */
/* #undef PJ_ANDROID */

### Log, call stack, etc

```shell
gcc -o ../bin/pjmedia-test-armv7l-unknown-linux-gnueabihf \
     output/pjmedia-test-armv7l-unknown-linux-gnueabihf/codec_vectors.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/jbuf_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/main.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/mips_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/vid_codec_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/vid_dev_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/vid_port_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/rtp_test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/test.o  output/pjmedia-test-armv7l-unknown-linux-gnueabihf/sdp_neg_test.o -lpjmedia-codec-armv7l-unknown-linux-gnueabihf -lpjmedia-videodev-armv7l-unknown-linux-gnueabihf -lpjmedia-audiodev-armv7l-unknown-linux-gnueabihf -lpjmedia-armv7l-unknown-linux-gnueabihf -lpjnath-armv7l-unknown-linux-gnueabihf -lpjlib-util-armv7l-unknown-linux-gnueabihf -lpj-armv7l-unknown-linux-gnueabihf -lsrtp-armv7l-unknown-linux-gnueabihf -lresample-armv7l-unknown-linux-gnueabihf -lgsmcodec-armv7l-unknown-linux-gnueabihf -lspeex-armv7l-unknown-linux-gnueabihf -lilbccodec-armv7l-unknown-linux-gnueabihf -lg7221codec-armv7l-unknown-linux-gnueabihf -lyuv-armv7l-unknown-linux-gnueabihf -lwebrtc-armv7l-unknown-linux-gnueabihf   -lssl -lcrypto -lvpx -lopenh264 -lstdc++ -luuid -lm -lrt -lpthread  -lasound -L/usr/local/lib -Wl,-rpath,/usr/local/lib -Wl,--enable-new-dtags -lSDL2   -lv4l2 -lopencore-amrnb -lopencore-amrwb   -L/root/pjproject-2.11/pjlib/lib -L/root/pjproject-2.11/pjlib-util/lib -L/root/pjproject-2.11/pjnath/lib -L/root/pjproject-2.11/pjmedia/lib -L/root/pjproject-2.11/pjsip/lib -L/root/pjproject-2.11/third_party/lib -L/usr/local/lib -Wl,-rpath,/usr/local/lib -Wl,--enable-new-dtags -lSDL2  -lv4l2  -lopenh264 -lstdc++ -lvpx -L/usr/local/lib -Wl,-rpath,/usr/local/lib -Wl,--enable-new-dtags -lSDL2  -lv4l2  -lopenh264 -lstdc++ -lvpx  -L/usr/local/lib -Wl,-rpath,/usr/local/lib -Wl,--enable-new-dtags -lSDL2  -lv4l2  -lopenh264 -lstdc++ -lvpx
/usr/bin/ld: /usr/local/lib/libpjmedia-videodev-armv7l-unknown-linux-gnueabihf.a(videodev.o): in function `pjmedia_vid_dev_subsys_init':
videodev.c:(.text+0x98): undefined reference to `pjmedia_dshow_factory'
collect2: error: ld returned 1 exit status
make[2]: *** [/root/pjproject-2.11/build/rules.mak:125: ../bin/pjmedia-test-armv7l-unknown-linux-gnueabihf] Error 1
make[2]: Leaving directory '/root/pjproject-2.11/pjmedia/build'
make[1]: *** [Makefile:245: pjmedia-test-armv7l-unknown-linux-gnueabihf] Error 2
make[1]: Leaving directory '/root/pjproject-2.11/pjmedia/build'
make: *** [Makefile:14: all] Error 1
```

**Comment by sauwming** [comment]
Follow the getting started guide in docs.pjsip.org.

**Comment by erwiower2** [comment]
Hello, were you able to fix this error? I’m currently stuck at the same point and can't resolve the issue with the documentation. The problem is that the sample project has an ActiveViewController, IncomingViewController, and OutgoingViewController.

I’ve made some changes to the Swift files, so the ActiveViewController no longer exists in its original function, and some of its functionality has been moved to the IncomingViewController. Because of this, I’m not sure if the issue with video calling not working is related to the changes I made.
